File: 009/009.js
function tripletProduct() {
  const s = 1000
  const r = Math.floor(Math.sqrt(s))
  for (let m = r; m > 1; m--)
    for (let n = m; n > 2; n--) {
      const a = m * m - n * n
      const b = 2 * (m * n)
      const c = m * m + n * n
      if (a + b + c === s) {
        console.log(a * b * c)
        return
      }
    }
}

tripletProduct()

File: 009/009.cc
#include <iostream>
#include <cmath>

int main() {
    int s = 1000;
    int r = (int) sqrt(s);
    for (int m = r; m > 1; m--)
        for (int n = m; n > 2; n--) {
            int a = m * m - n * n;
            int b = 2 * (m * n);
            int c = m * m + n * n;
            if (a + b + c == s) {
                std::cout << a * b * c << std::endl;
                return 0;
            }
        }
}
